experimenter/experimenter/nimbus-ui/src/components/Summary/TableAudience/index.tsx

```diff
@@ -80,11 +83,67 @@ const TableAudience = ({ experiment }: TableAudienceProps) => {
-                        ))}
-                      </ul>
+                      experiment.locales.length > 10 ? (
+                        <>
+                          <Accordion>
+                            <Accordion.Toggle
+                              as={Accordion}
+                              eventKey="0"
+                              onClick={() => setExpandLocale(!expandLocale)}
+                            >
+                              {expandLocale ? (
+                                <>
+                                  <div className="float-right">
+                                    <Button
+                                      size="sm"
+                                      variant="outline-primary"
+                                      data-testid="locales-hide"
+                                    >
+                                      <CollapseMinus />
+                                      Hide
+                                    </Button>
+                                  </div>
+                                </>
+                              ) : (
+                                <>
+                                  <div className="float-right">
+                                    <Button
+                                      size="sm"
+                                      variant="outline-primary"
+                                      data-testid="locales-show-more"
+                                    >
+                                      <ExpandPlus />
+                                      Show More
+                                    </Button>
+                                  </div>
+                                  <ul className="list-unstyled mb-0">
+                                    {experiment.locales
+                                      .slice(0, 10)
+                                      .map((l) => (
+                                        <li key={l.id}>{l.name}</li>
+                                      ))}{" "}
+                                    ...
+                                  </ul>
+                                </>
+                              )}
+                            </Accordion.Toggle>
+
+                            <Accordion.Collapse eventKey="0">
+                              <ul className="list-unstyled mb-0">
+                                {experiment.locales.map((l) => (
+                                  <li key={l.id}>{l.name}</li>
+                                ))}
+                              </ul>
+                            </Accordion.Collapse>
+                          </Accordion>
+                        </>
+                      ) : (
+                        <ul className="list-unstyled mb-0">
+                          {experiment.locales.map((l) => (
+                            <li key={l.id}>{l.name}</li>
+                          ))}
+                        </ul>
+                      )
```

Comment:
Could this be its own component since it's now a non-trivial amount of code that is basically copied three times in this file?

Reply:
good point, refactored it